plugin/web/extensions/sidebar/core/components/content/common/DatasetCard/Field/Fields/general/StyleCode/hooks.ts

```diff
@@ -1,48 +1,32 @@
-import { postMsg } from "@web/extensions/sidebar/utils";
 import JSON5 from "json5";
 import { useCallback, useEffect, useState } from "react";
 
 import { BaseFieldProps } from "../../types";
 
-export default ({
-  value,
-  dataID,
-  onUpdate,
-}: Pick<BaseFieldProps<"styleCode">, "value" | "dataID" | "onUpdate">) => {
+export default ({ value, onUpdate }: Pick<BaseFieldProps<"styleCode">, "value" | "onUpdate">) => {
   const [code, editCode] = useState(value.src);
 
-  const updateStyle = useCallback(
-    (styleStr: string) => {
-      if (styleStr) {
-        try {
-          const styleObject = JSON5.parse(styleStr);
-          postMsg({ action: "updateDatasetInScene", payload: { dataID, update: styleObject } });
-          // eslint-disable-next-line no-empty
-        } catch (error) {}
-      }
-    },
-    [dataID],
-  );
-
   const onApply = useCallback(() => {
-    updateStyle(code);
-  }, [updateStyle, code]);
-
-  const onEdit = useCallback(
-    (e: React.ChangeEvent<HTMLTextAreaElement>) => {
-      editCode(e.target.value);
+    try {
+      const styleObject = JSON5.parse(code);
       onUpdate({
         ...value,
-        src: e.target.value,
+        src: code,
+        override: styleObject,
       });
-    },
-    [onUpdate, value],
-  );
+      // eslint-disable-next-line no-empty
+    } catch (error) {}
+  }, [onUpdate, code, value]);
 
   useEffect(() => {
-    updateStyle(value.src);
+    onApply();
```

Comment:
Is this for initialization? If so, it shouldn't be needed since you pass an override field in the onUpdate.